sdk/search/azure-search/src/main/java/com/azure/search/models/SearchResult.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.azure.search.Document;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Contains a document found by a search query, plus associated metadata.
+ */
+@Fluent
+public final class SearchResult {
+    /*
+     * Unmatched properties from the message are deserialized this collection
+     */
+    @JsonProperty(value = "")
+    private Document additionalProperties;
+
+    /*
+     * The relevance score of the document compared to other documents returned
+     * by the query.
+     */
+    @JsonProperty(value = "@search.score", access = JsonProperty.Access.WRITE_ONLY)
+    private double score;
+
+    /*
+     * Text fragments from the document that indicate the matching search
+     * terms, organized by each applicable field; null if hit highlighting was
+     * not enabled for the query.
+     */
+    @JsonProperty(value = "@search.highlights", access = JsonProperty.Access.WRITE_ONLY)
+    private Map<String, List<String>> highlights;
+
+    /**
+     * Get the additionalProperties property: Unmatched properties from the
+     * message are deserialized this collection.
+     *
+     * @return the additionalProperties value.
+     */
+    @JsonIgnore
+public Document getDocument() {
+        return this.additionalProperties;
+    }
+
+    /**
+     * Set the additionalProperties property: Unmatched properties from the
+     * message are deserialized this collection.
+     *
+     * @param additionalProperties the additionalProperties value to set.
+     * @return the SearchResult object itself.
+     */
+    SearchResult setDocument(Document additionalProperties) {
```

Comment:
I think this is the result of another rogue regex rule, since all other properties of this class are immutable. This should be too. We need to fix this before preview.